linera-core/src/client.rs

```diff
@@ -1070,6 +1166,34 @@ where
         }
     }
 
+    /// Tries to read blobs from either the pending blobs or the local node's cache, or
+    /// the chain manager's pending blobs
+    async fn read_local_blobs(
+        &mut self,
+        blob_ids: impl IntoIterator<Item = BlobId>,
+        chain_managers_pending_blobs: &BTreeMap<BlobId, HashedBlob>,
+    ) -> Result<Vec<HashedBlob>, LocalNodeError> {
+        let mut blobs = Vec::new();
+        for blob_id in blob_ids {
+            if let Some(blob) = self.node_client.recent_blob(&blob_id).await {
+                blobs.push(blob);
+            } else if let Some(blob) = self.pending_blobs.get(&blob_id) {
+                self.node_client.cache_recent_blob(blob).await;
+                blobs.push(blob.to_owned());
+            } else if let Some(blob) = chain_managers_pending_blobs.get(&blob_id) {
+                self.node_client.cache_recent_blob(blob).await;
+                blobs.push(blob.to_owned());
```

Comment:
Do we need this here? This is before we even handle the block proposal in our local node, and thus in the chain manager, so usually I'd expect it to not be there, and to only be in the client's `pending_blobs`?

(In general, it would be good to clarify when a blob is supposed to be where.)

Reply:
Ah right, because we reset the chain manager when execute a block. Should I clarify that with comment around the code? Or is there another specific way that should be better?

Reply:
Maybe as comments on the respective fields, like `pending_blobs`: That this contains blobs belonging to our `pending_block` that may not even have processed by our own chain manager yet.

And in the chain manager, that these are blobs belonging to proposed or validated blocks that have not been confirmed yet.